pkg/types/urls.go

```diff
@@ -39,11 +39,26 @@ func NewURLs(strs []string) (URLs, error) {
 		if u.Scheme != "http" && u.Scheme != "https" && u.Scheme != "unix" && u.Scheme != "unixs" {
 			return nil, fmt.Errorf("URL scheme must be http, https, unix, or unixs: %s", in)
 		}
-		if _, _, err := net.SplitHostPort(u.Host); err != nil {
-			return nil, fmt.Errorf(`URL address does not have the form "host:port": %s`, in)
-		}
-		if u.Path != "" {
-			return nil, fmt.Errorf("URL must not contain a path: %s", in)
+		switch u.Scheme {
+		case "http":
+			fallthrough
+		case "https":
+			if _, _, err = net.SplitHostPort(u.Host); err != nil {
+				return nil, fmt.Errorf(`URL address does not have the form "host:port": %s`, in)
+			}
+			if u.Path != "" {
+				return nil, fmt.Errorf("URL must not contain a path: %s", in)
+			}
+		case "unix":
+			fallthrough
+		case "unixs":
+			if u.Host != "" {
```

Comment:
> and net.Listen(network, addr) in embed/etcd.go:configureClientListeners

I think the `configureClientListeners` works without modifying `u.Host` already.

It sets  `addr = u.Host + u.Path` before the `net.Listen(network, addr)`, since commit 4f27981c46d6d4eb42c803ecc97e7fcdf9e8c2d5